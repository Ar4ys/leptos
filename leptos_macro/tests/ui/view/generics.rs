use leptos::*;

#[component]
fn Component<T: Into<String>>(prop: T) -> impl IntoView {
    _ = prop;
}

fn main() {
    view! {
        <Component<i32> prop=0 />
    };

    view! {
        <Component prop=0 />
    };
}
