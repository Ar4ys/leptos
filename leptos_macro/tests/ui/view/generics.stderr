error[E0277]: the trait bound `std::string::String: From<i32>` is not satisfied
  --> tests/ui/view/generics.rs:10:20
   |
10 |         <Component<i32> prop=0 />
   |                    ^^^ the trait `From<i32>` is not implemented for `std::string::String`
   |
   = help: the following other types implement trait `From<T>`:
             <std::string::String as From<char>>
             <std::string::String as From<Box<str>>>
             <std::string::String as From<Oco<'_, str>>>
             <std::string::String as From<Cow<'a, str>>>
             <std::string::String as From<url::Url>>
             <std::string::String as From<js_sys::JsString>>
             <std::string::String as From<&'a js_sys::JsString>>
             <std::string::String as From<&str>>
           and $N others
   = note: required for `i32` to implement `Into<std::string::String>`
note: required by a bound in `Component`
  --> tests/ui/view/generics.rs:4:17
   |
4  | fn Component<T: Into<String>>(prop: T) -> impl IntoView {
   |                 ^^^^^^^^^^^^ required by this bound in `Component`

error[E0277]: the trait bound `fn(ComponentProps<i32>) -> impl leptos::IntoView {Component::<i32>}: leptos::Component<_>` is not satisfied
  --> tests/ui/view/generics.rs:10:10
   |
10 |         <Component<i32> prop=0 />
   |          ---------^^^^^
   |          |
   |          the trait `leptos::Component<_>` is not implemented for fn item `fn(ComponentProps<i32>) -> impl leptos::IntoView {Component::<i32>}`
   |          required by a bound introduced by this call
   |
note: required by a bound in `component_props_builder`
  --> $WORKSPACE/leptos/src/lib.rs
   |
   | pub fn component_props_builder<P: PropsOrNoPropsBuilder>(
   |        ----------------------- required by a bound in this function
   |     _f: &impl Component<P>,
   |               ^^^^^^^^^^^^ required by this bound in `component_props_builder`

error[E0277]: the trait bound `std::string::String: From<i32>` is not satisfied
  --> tests/ui/view/generics.rs:10:10
   |
10 |         <Component<i32> prop=0 />
   |          ^^^^^^^^^ the trait `From<i32>` is not implemented for `std::string::String`
   |
   = help: the following other types implement trait `From<T>`:
             <std::string::String as From<char>>
             <std::string::String as From<Box<str>>>
             <std::string::String as From<Oco<'_, str>>>
             <std::string::String as From<Cow<'a, str>>>
             <std::string::String as From<url::Url>>
             <std::string::String as From<js_sys::JsString>>
             <std::string::String as From<&'a js_sys::JsString>>
             <std::string::String as From<&str>>
           and $N others
   = note: required for `i32` to implement `Into<std::string::String>`
note: required by a bound in `Component`
  --> tests/ui/view/generics.rs:4:17
   |
4  | fn Component<T: Into<String>>(prop: T) -> impl IntoView {
   |                 ^^^^^^^^^^^^ required by this bound in `Component`

error[E0277]: the trait bound `std::string::String: From<{integer}>` is not satisfied
  --> tests/ui/view/generics.rs:14:20
   |
14 |         <Component prop=0 />
   |                    ^^^^ the trait `From<{integer}>` is not implemented for `std::string::String`
   |
   = help: the following other types implement trait `From<T>`:
             <std::string::String as From<char>>
             <std::string::String as From<Box<str>>>
             <std::string::String as From<Oco<'_, str>>>
             <std::string::String as From<Cow<'a, str>>>
             <std::string::String as From<url::Url>>
             <std::string::String as From<js_sys::JsString>>
             <std::string::String as From<&'a js_sys::JsString>>
             <std::string::String as From<&str>>
           and $N others
   = note: required for `{integer}` to implement `Into<std::string::String>`
note: required by a bound in `ComponentPropsBuilder::<T>::prop`
  --> tests/ui/view/generics.rs:4:17
   |
4  | fn Component<T: Into<String>>(prop: T) -> impl IntoView {
   |                 ^^^^^^^^^^^^ required by this bound in `ComponentPropsBuilder::<T>::prop`

error[E0277]: the trait bound `std::string::String: From<{integer}>` is not satisfied
  --> tests/ui/view/generics.rs:14:10
   |
14 |         <Component prop=0 />
   |          ^^^^^^^^^ the trait `From<{integer}>` is not implemented for `std::string::String`
   |
   = help: the following other types implement trait `From<T>`:
             <std::string::String as From<char>>
             <std::string::String as From<Box<str>>>
             <std::string::String as From<Oco<'_, str>>>
             <std::string::String as From<Cow<'a, str>>>
             <std::string::String as From<url::Url>>
             <std::string::String as From<js_sys::JsString>>
             <std::string::String as From<&'a js_sys::JsString>>
             <std::string::String as From<&str>>
           and $N others
   = note: required for `{integer}` to implement `Into<std::string::String>`
note: required by a bound in `Component`
  --> tests/ui/view/generics.rs:4:17
   |
4  | fn Component<T: Into<String>>(prop: T) -> impl IntoView {
   |                 ^^^^^^^^^^^^ required by this bound in `Component`

error[E0599]: the method `build` exists for struct `ComponentPropsBuilder<{integer}, (({integer},),)>`, but its trait bounds were not satisfied
  --> tests/ui/view/generics.rs:14:10
   |
3  | #[component]
   | ------------ method `build` not found for this struct
...
14 |         <Component prop=0 />
   |          ^^^^^^^^^ method cannot be called on `ComponentPropsBuilder<{integer}, (({integer},),)>` due to unsatisfied trait bounds
   |
note: trait bound `{integer}: Into<std::string::String>` was not satisfied
  --> tests/ui/view/generics.rs:3:1
   |
3  | #[component]
   | ^^^^^^^^^^^^ unsatisfied trait bound introduced in this `derive` macro
   = note: this error originates in the attribute macro `component` (in Nightly builds, run with -Z macro-backtrace for more info)
